midterm-2023-11-06/code/5184-leona-kirijas/script.ts

```diff
@@ -0,0 +1,189 @@
+
+    const countryElement = document.getElementById("country-filter") as HTMLSelectElement;
+    const country = countryElement.value;
+
+    const genreElement = document.getElementById("genre-filter") as HTMLSelectElement;
+    const genre = bandNameElement.value.toLowerCase();
+    let filteredBands = bands;
+
+    if (bandName) {
+        filteredBands = filteredBands.filter(band => band.name.toLowerCase().includes(bandName));
+    }
+
+    if (country !== "all") {
+        filteredBands = filteredBands.filter(band => band.country === country);
+    }
+
+    displayBands(filteredBands);
+}
+const loadData = async () => {
+    const dataUri = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json";
+    const response = await fetch(dataUri);
+
+    if (!response.ok) {
+        throw new Error("The data is not there");
+    }
+
+    const data = await response.json();
+    return data;
+
+
+}
+
+
+const displayBands = (bands: Band[]) => {
+    const container = document.getElementById("band-container")!;
+    container.innerHTML = "";
+    for (const band of bands) {
+        const bandRow = generateBandRow(band);
+        container.appendChild(bandRow as any);
+    }
+}
+
+const generateBandRow = (band: Band) => {
+    const row = document.createElement("div");
+    row.classList.add("band-row");
+
+    // id cell
+    const idCell = document.createElement("div");
+    idCell.classList.add("band-data", "band-id");
+    idCell.innerHTML = band.id.toString();
+    row.appendChild(idCell);
+
+    const nameCell = document.createElement("div");
+    nameCell.classList.add("band-data", "band-name");
+    nameCell.innerHTML = `${band.name} / ${band.genre}`;
+    row.appendChild(nameCell);
+
+    const formedCell = document.createElement("div");
+    formedCell.classList.add("band-data", "band-formed");
+    formedCell.innerHTML = band.formed.toString();
+    row.appendChild(formedCell);
+
+    const locationCell = document.createElement("div");
+    locationCell.classList.add("band-data", "band-location");
+    locationCell.innerHTML = band.location;
+    row.appendChild(locationCell);
+
+    const locationCell1 = document.createElement("div");
+    locationCell1.classList.add("band-data", "band-location");
+    locationCell1.innerHTML = band.location;
+    row.appendChild(locationCell1);
+
+    const countryCell = document.createElement("div");
+    countryCell.classList.add("band-data", "band-location");
+    countryCell.innerHTML = band.country;
+    row.appendChild(countryCell);
+
+    // const genreCell = document.createElement("div");
+    // genreCell.classList.add("band-data", "band-location");
+    // genreCell.innerHTML = band.genre;
+    // row.appendChild(countryCell);
+
+    const membersCell = document.createElement("div");
+    membersCell.classList.add("band-data", "band-members");
+    membersCell.innerHTML = band.members.join(", ");
```

Comment:
The band members are joined, but no special handling is done for the last member